### DISCUSSION TITLE
Can mrgsim_q() have an option to change maxsteps?

### DISCUSSION THREAD

**Discussion by annabel-meade** [metrumresearchgroup/mrgsolve#1083] [Q&A] [answered]
I am trying to optimize parameters, so I would like to use the faster mrgsim_q() over mrgsim(). However, my model spits an error if I do not increase the maxsteps from 20,000 to 500,000. Is there a way to change maxsteps in mrgsim_q(), or am I misunderstanding the purpose of this function?

**Comment by kylebaron** [comment, marked as answer]
Hi @annabel-meade  - 

The different solver settings are stored in the model object. You can change them through the `update()` method. See the example here:  https://mrgsolve.org/user-guide/components.html#solver-settings

The difference here with `mrgsim()` and `mrgsim_q()` is that `mrgsim()` has a `...` argument so you can pass arguments through `...` to `update()` as a part of the call to `mrgsim()`; it's old functionality and probably a little confusing but we keep it for convenience. 

So if you're writing this `mrgsim()` code:

```r
mod %>% mrgsim(maxsteps = 500000)
```

just change it to this:

```r
mod <- update(mod, maxsteps = 500000)

mod %>% mrgsim_q()

```

Hope this helps; please let me know if I've misunderstood your question.

Kyle

**Reply by annabel-meade** [reply]
Thank you, this helped a lot!
